doc/src/guides/solving/solve-equation-algebraically.md

```diff
@@ -0,0 +1,449 @@
+>>> x = Symbol('x', real=True)
+>>> expr = (x-4)*(x-3)*(x-2)*(x-1)
+>>> solution = solve(expr, x)
+>>> print(solution)
+[1, 2, 3, 4]
+>>> solution_outside_2_3 = [v for v in solution if (v.is_real and Or(v<2,v>3))]
+>>> print(solution_outside_2_3)
+[1, 4]
+>>> solution_2_3 = solve((expr,x>=2,x<=3), x)
+>>> print(solution_2_3)
+Eq(x, 2) | Eq(x, 3)
+```
+
+For {func}`~.solveset`, limit the output domain in the function call by setting a domain
+
+```py
+>>> from sympy import S, solveset
+>>> from sympy.abc import x
+>>> solution = solveset(x**4 - 256, x, domain=S.Reals)
+>>> print(solution)
+{-4, 4}
+```
+
+or by restricting returned solutions to any arbitrary set, including an interval:
+
+```py
+>>> from sympy import Interval, pi, sin, solveset
+>>> from sympy.abc import x
+>>> solution = solveset(sin(x), x, Interval(-pi, pi))
+>>> print(solution)
+{0, -pi, pi}
+```
+
+and if you restrict the solutions to a domain in which there are no solutions, {func}`~.solveset` will return the empty set, [EmptySet](../../modules/sets.rst):
+
+```py
+>>> from sympy import solveset, S
+>>> from sympy.abc import x
+>>> solution = solveset(x**2 + 1, x, domain=S.Reals)
+>>> print(solution)
+EmptySet
+```
+
+### Explicitly represent infinite sets of possible solutions using {func}`~.solveset`
+
+{func}`~.solveset` 
+[can represent infinite sets of possible solutions](why-solveset)
+and express them in standard mathematical notation, 
+for example $\sin(x) = 0$ for $x = n * \pi$ for every integer value of $n$:
+
+```py
+>>> from sympy import pprint, sin, solveset
+>>> from sympy.abc import x
+>>> solution = solveset(sin(x), x)
+>>> pprint(solution)
+{2*n*pi | n in Integers} U {2*n*pi + pi | n in Integers}
+```
+
+However, {func}`~.solve` will return only a finite number of solutions:
+
+```py
+>>> from sympy import sin, solve
+>>> from sympy.calculus.util import periodicity
+>>> from sympy.abc import x
+>>> f = sin(x)
+>>> solution = solve(f, x)
+>>> print(solution)
+[0, pi]
+>>> print(periodicity(f, x))
+2*pi
+```
+
+{func}`~.solve` tries to return just enough solutions so that all (infinitely many) solutions can generated 
+from the returned solutions by adding integer multiples of the {func}`~.periodicity` of the equation, here $2\pi$.
+
+## Use the solution result
+
+### Substitute solutions from {func}`~.solve` into an expression
+
+You can substitute solutions from {func}`~.solve` into an expression.
+
+A common use case is finding the critical points and values for a function $f$. 
+At the critical points, the {class}`~.Derivative` equals zero (or is undefined). 
+You can then obtain the function values at those critical points
+by substituting the critical points back into the function using {meth}`~sympy.core.basic.Basic.subs`. You can also tell if the critical point is a maxima or minima by substituting the values
+into the expression for the second derivative: a negative value indicates a maximum, and a positive value indicates a minimum.
```

Comment:
Long lines should be wrapped. This is something that I do with my editor. I don't know what editor you use but I use vim and I set the textwidth to 78. Then I can highlight a paragraph like this use `:gq` to wrap the lines to that width. Other editors will have ways of doing the same thing.

Reply:
I'm using Visual Studio Code and just installed the ReWrap extension to do that by pressing Option-Q on my Mac (Alt-Q on Windows).